core/utils/cuckoo_map.h

```diff
@@ -173,26 +174,24 @@ class CuckooMap {
   iterator begin() { return iterator(*this, 0, 0); }
   iterator end() { return iterator(*this, buckets_.size(), 0); }
 
-  // Insert/update a key value pair
-  // Return the pointer to the inserted entry
-  Entry* Insert(const K& key, const V& value, const H& hasher = H(),
-                const E& eq = E()) {
+  template <typename... Args>
+  Entry* DoEmplace(const K& key, const H& hasher, const E& eq, Args&&... args) {
     Entry* entry;
     HashResult primary = Hash(key, hasher);
 
     EntryIndex idx = FindWithHash(primary, key, eq);
     if (idx != kInvalidEntryIdx) {
       entry = &entries_[idx];
-      entry->second = value;
+      new (&entry->second) V(std::forward<Args>(args)...);
```

Comment:
General note-to-the-universe (not an issue with this change): I hate placement-new. 😝